tests/core/arch-arm64/stalker-arm64.c

```diff
@@ -346,6 +352,251 @@ store_x0 (GumCpuContext * cpu_context,
+
+    /* func_b: */
+    0xd2801320, /* mov x0, #0x99  */
+    0xd65f03c0, /* ret            */
+  };
+
+  StalkerTestFunc func;
+  guint8 * func_a_address;
+
+  fixture->sink->mask = GUM_EXEC;
+  g_assert_cmpuint (fixture->sink->events->len, ==, 0);
+
+  func = GUM_POINTER_TO_FUNCPTR (StalkerTestFunc,
+      test_arm64_stalker_fixture_dup_code (fixture, code_template,
+        sizeof (code_template)));
+
+  func_a_address = fixture->code + (16 * 4);
+
+  GumMemoryRange memory_range;
+  memory_range.base_address = (GumAddress) func_a_address;
+  memory_range.size = (4 * 2);
+
+  gum_stalker_exclude (fixture->stalker, &memory_range);
+
+  test_arm64_stalker_fixture_follow_and_invoke (fixture, func, 0);
+
+  g_assert_cmpuint (fixture->sink->events->len, ==, 24);
+}
+
+STALKER_TESTCASE (exclude_blr)
+{
+  StalkerTestFunc func;
+  guint8 * code;
+  GumArm64Writer cw;
+  gint r;
+  gpointer func_a;
+  const gchar * start_lbl = "start";
+
+  fixture->sink->mask = GUM_EXEC;
+  g_assert_cmpuint (fixture->sink->events->len, ==, 0);
+
+  code = gum_alloc_n_pages (1, GUM_PAGE_RWX);
+  gum_arm64_writer_init (&cw, code);
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);
+  gum_arm64_writer_put_call_address_with_arguments (&cw,
+      GUM_ADDRESS (gum_stalker_follow_me), 3,
+      GUM_ARG_ADDRESS, GUM_ADDRESS (fixture->stalker),
+      GUM_ARG_ADDRESS, GUM_ADDRESS (fixture->transformer),
+      GUM_ARG_ADDRESS, GUM_ADDRESS (fixture->sink));
+  gum_arm64_writer_put_pop_all_x_registers (&cw);
+
+  gum_arm64_writer_put_b_label (&cw, start_lbl);
+
+  func_a = gum_arm64_writer_cur (&cw);
+  gum_arm64_writer_put_add_reg_reg_imm (&cw, ARM64_REG_X0, ARM64_REG_X0, 10);
+  gum_arm64_writer_put_ret (&cw);
+
+  gum_arm64_writer_put_label (&cw, start_lbl);
+  gum_arm64_writer_put_push_reg_reg (&cw, ARM64_REG_X19, ARM64_REG_LR);
+  gum_arm64_writer_put_ldr_reg_address (&cw, ARM64_REG_X1, GUM_ADDRESS (func_a));
+  gum_arm64_writer_put_blr_reg (&cw, ARM64_REG_X1);
+  gum_arm64_writer_put_pop_reg_reg (&cw, ARM64_REG_X19, ARM64_REG_LR);
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);
+  gum_arm64_writer_put_call_address_with_arguments (&cw,
+      GUM_ADDRESS (gum_stalker_unfollow_me), 1,
+      GUM_ARG_ADDRESS, GUM_ADDRESS (fixture->stalker));
+  gum_arm64_writer_put_pop_all_x_registers (&cw);
+
+  gum_arm64_writer_put_ret (&cw);
+
+  gum_arm64_writer_flush (&cw);
+  gum_clear_cache (cw.base, gum_arm64_writer_offset (&cw));
+  gum_arm64_writer_clear (&cw);
+
+  GumMemoryRange memory_range;
+  memory_range.base_address = GUM_ADDRESS (func_a);
+  memory_range.size = (4 * 2);
+
+  gum_stalker_exclude (fixture->stalker, &memory_range);
+
+  func = GUM_POINTER_TO_FUNCPTR (StalkerTestFunc, code);
+  r = func (2);
+
+  g_assert_cmpint (r, ==, 12);
+
+  g_assert_cmpuint (fixture->sink->events->len, ==, 42);
+
+  gum_free_pages (code);
+}
+
+STALKER_TESTCASE (exclude_bl_with_unfollow)
+{
+  StalkerTestFunc func;
+  guint8 * code;
+  GumArm64Writer cw;
+  gint r;
```

Comment:
Let's avoid single-letter variable names. Maybe name it `result` or something descriptive.